llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2859,26 +2859,23 @@ static Value *stripGetElementPtr(Value *Ptr, ScalarEvolution *SE, Loop *Lp) {
 /// strides "a[i*stride]". Returns the symbolic stride, or null otherwise.
 static const SCEV *getStrideFromPointer(Value *Ptr, ScalarEvolution *SE, Loop *Lp) {
   auto *PtrTy = dyn_cast<PointerType>(Ptr->getType());
-  if (!PtrTy || PtrTy->isAggregateType())
+  if (!PtrTy)
```

Comment:
It's not obvious to me why this is NFC. Even if it's not possible to get here with aggregate types currently it still feels wrong to mark it as NFC since technically this function now permits aggregates. Can you document the reasoning behind this change in the commit message. Not sure what @fhahn thinks, but I'd prefer removing NFC from the title too. Assuming this change is safe, it would be good to add an assert here that the pointer type is not an aggregate type.

Reply:
I think it definitely should be an assert; I checked on a large test set and it doesn't trigger, but we don't want the code to silently do the wrong thing in the future when new users are added.

Reply:
Ah yes that makes sense

Reply:
(although even without opaque pointers, I think it could either have been a pointer or aggregate type)